### ISSUE TITLE
SMB mount error(13): Permission denied

### ISSUE THREAD

**Issue by wdixon** [hashicorp/vagrant#8620]
### Vagrant version
Vagrant 1.9.5

### Host operating system
Windows 7 (64-bit)

### Guest operating system
Ubuntu 16.04

### Vagrantfile
```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "get-mca/dev"
  config.vm.box_url = "file://vm.json"
  config.ssh.forward_x11 = true
  config.ssh.forward_agent = true
  config.vm.box_check_update = false
  config.vm.synced_folder ".", "/vagrant"

  config.vm.provider "virtualbox" do |v|
    v.check_guest_additions = false
    v.functional_vboxsf     = false

    v.gui = false
    v.name = "getmca_dev"
    v.customize ["modifyvm", :id, "--memory", "1024"]
    v.customize ["modifyvm", :id, "--vram", "12"]
    v.customize ["modifyvm", :id, "--cpus", "1"]
    v.customize ["modifyvm", :id, "--cpuexecutioncap", "50"]
  end
end

```

### Debug output
==> default: Machine booted and ready!
==> default: Configuring proxy for Apt...
==> default: Configuring proxy for Docker...
==> default: Configuring proxy environment variables...
==> default: Mounting SMB shared folders...
    default: C:/cygwin64/home/200003581/GET-MCA/VirtualMachines/packer/example => /vagrant
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t cifs -o uid=1000,gid=1000,sec=ntlm,credentials=/etc/smb_creds_e18c37823b835d29d892174d053c9a9c //10.10.10.1/e18c37823b835d29d892174d053c9a9c /vagrant

The error output from the last command was:

mount error(13): Permission denied
Refer to the mount.cifs(8) manual page (e.g. man mount.cifs)

### Expected behavior
When the smb mount is attempted on the guest, it unconditionally inserts "sec=ntlm" 
However, this mode is not allowed in all environments.  The newer cifs package would default to ntlmssp, but is overridden in the vagrant smb plugin 

HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.9.5/plugins/guests/linux/cap/mount_smb_shared_folder.rb

If I comment out the ntlm option
```
#  options[:mount_options] << "sec=ntlm"
```
or replace with ntlmssp:
```
  options[:mount_options] << "sec=ntlmssp"
```

I am able to establish the SMB share...  
The current code does not allow this option to be overridden the configuration.  So even if you supplied the sec setting as a mount option:

  config.vm.synced_folder ".", "/vagrant", mount_options:["sec=ntlmssp"]

you end up with both the request and the forced sec option:

mount -t cifs -o uid=1000,gid=1000,sec=ntlm,credentials=/etc/smb_creds_e18c37823b835d29d892174d053c9a9c //<MY_IP_ADDRESS>/e18c37823b835d29d892174d053c9a9c /vagrant

Which causes the mount to fail

It would be good to only insert sec=ntlm if the sec option was not specified in Vagrantfile mount_options.

### Actual behavior
What actually happened?

### Steps to reproduce
1.
2.
3.

### References
Are there any other GitHub issues (open or closed) that should be linked here?
For example:
- Possibly GH-8381 - similar to and perhaps the root cause but was closed for other reason
- GH-4823 - but the fix has not made it into the main branch (now marked closed).

**Comment by TonyApuzzo** [comment]
I added PR #9170 that addresses this issue by allowing the Vagrant file to override the Vagrant hard-coded ntlm.

**Comment by chathunpc** [comment]
Make sure disable or allow your firewall and antivirus software to allow network transaction I spent many hours to get this :)

**Comment by fernandogmo** [comment]
To anybody else facing the exact error message @wdixon got, this is what helped me:

1. The username being asked for is the one you get when you run `$Env:USERNAME` in PowerShell.
2. The password being asked for is the one you would use for login into the **current Windows user account** on that host machine. This _may_ differ from your Microsoft account password if you have recently changed that one (this is the case for me).

A fairly simple fix (in my case) so I hope this helps!

**Comment by jbonhag** [comment]
Hi there,

It looks like this has been resolved by #9366 so I am now closing this issue. If the original issue was not fully resolved, please reopen this issue or create a new one.

Cheers!
